src/app/pages/personal-details/personal-details.component.ts

```diff
@@ -1,26 +1,40 @@
-import { Component, Input } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Observable, switchMap } from 'rxjs';
 import { DetailsPersonService } from 'src/app/features/person/services/details-person.service';
 import { DetailsPersonData, PersonResult } from 'src/app/features/person/types';
 
+
+const DEFAULT_DATA:DetailsPersonData = { type: 'person', id: 123 }
+
 @Component({
   selector: 'mov-personal-details',
   templateUrl: './personal-details.component.html',
   styleUrls: ['./personal-details.component.scss']
 })
-export class PersonalDetailsComponent {
+export class PersonalDetailsComponent implements OnInit {
   constructor(
     private readonly apiDetailsMedia: DetailsPersonService,
+    private readonly route: ActivatedRoute
   ) {
+  }
 
-    this.apiDetailsMedia.requestDetailsByType<PersonResult>(this.data)
-      .subscribe((detail:PersonResult) => {
-        this.detailData = detail
-        console.log(this.detailData)
-      }
-    )
+  public data:DetailsPersonData = {
+    type: DEFAULT_DATA.type,
+    id: DEFAULT_DATA.id
   }
+  public personDetails$?: Observable<PersonResult>
 
-  @Input() public data:DetailsPersonData = {type: 'person', id: 123 }
+  ngOnInit(): void {
+    this.personDetails$ = this.route.paramMap.pipe(
+      switchMap(params => {
+        this.data.id = Number(params.get('id')) || DEFAULT_DATA.id
+        return this.getPersonDetails(this.data)
+      })
+    )
+  }
```

Comment:
Same issues as in media-details component except type.

I think you can store this type inside the 'getPersonDetails' method, because it is not dynamic value.